htdocs/comm/action/index.php

```diff
@@ -919,7 +919,7 @@
 // Complete $eventarray with events coming from external module
 $parameters=array(); $object=null;
 $reshook=$hookmanager->executeHooks('getCalendarEvents',$parameters,$object,$action);
-if (! empty($hookmanager->resArray['eventarray'])) $eventarray=array_merge($eventarray, $hookmanager->resArray['eventarray']);
+if (! empty($hookmanager->resArray['eventarray'])) $eventarray=array_replace($eventarray, $hookmanager->resArray['eventarray']);
```

Comment:
That was my first idea, but this fix change the behavior of the hook, and exisiting modules can work with the 'replace' behavior but not the 'merge'.

Maybe `array_replace` in 3.8, 3.9, 4 and the merge loop in 5+ ?